File: core/src/test/java/com/jetbrains/tmp/learning/courseFormat/StudyNodeTest.java
package com.jetbrains.tmp.learning.courseFormat;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * @author meanmail
 */
public class StudyNodeTest {

    private CourseNode courseNode;
    private SectionNode sectionNode;
    private LessonNode lessonNode;
    private StepNode stepNode;

    @Before
    public void setUp() throws Exception {
        courseNode = new CourseNode();

        sectionNode = new SectionNode();
        sectionNode.setId(1);
        sectionNode.setParent(courseNode);


        lessonNode = new LessonNode();
        lessonNode.setId(1);
        lessonNode.setParent(sectionNode);

        stepNode = new StepNode();
        stepNode.setId(1);
        stepNode.setParent(lessonNode);

        courseNode.getChildren().add(sectionNode);
        sectionNode.getChildren().add(lessonNode);
        lessonNode.getChildren().add(stepNode);
    }

    @Test
    public void getCourseDirectory() throws Exception {
        assertEquals("", courseNode.getDirectory());
    }

    @Test
    public void getCoursePath() throws Exception {
        assertEquals("", courseNode.getPath());
    }

    @Test
    public void getSectionDirectory() throws Exception {
        assertEquals("section1", sectionNode.getDirectory());
    }

    @Test
    public void getSectionPath() throws Exception {
        assertEquals("section1", sectionNode.getPath());
    }

    @Test
    public void getLessonDirectory() throws Exception {
        assertEquals("lesson1", lessonNode.getDirectory());
    }

    @Test
    public void getLessonPath() throws Exception {
        assertEquals("section1/lesson1", lessonNode.getPath());
    }

    @Test
    public void getStepDirectory() throws Exception {
        assertEquals("step1", stepNode.getDirectory());
    }

    @Test
    public void getStepPath() throws Exception {
        assertEquals("section1/lesson1/step1", stepNode.getPath());
    }

    @Test
    public void getSectionDirectoryAfterChangeId() throws Exception {
        sectionNode.setId(2);
        assertEquals("section2", sectionNode.getDirectory());
    }

    @Test
    public void getSectionPathAfterChangeId() throws Exception {
        sectionNode.setId(2);
        assertEquals("section2", sectionNode.getPath());
    }

    @Test
    public void getLessonDirectoryAfterChangeId() throws Exception {
        lessonNode.setId(2);
        assertEquals("lesson2", lessonNode.getDirectory());
    }

    @Test
    public void getLessonPathAfterChangeId() throws Exception {
        lessonNode.setId(2);
        assertEquals("section1/lesson2", lessonNode.getPath());
    }

    @Test
    public void getStepDirectoryAfterChangeId() throws Exception {
        stepNode.setId(2);
        assertEquals("step2", stepNode.getDirectory());
    }

    @Test
    public void getStepPathAfterChangeId() throws Exception {
        stepNode.setId(2);
        assertEquals("section1/lesson1/step2", stepNode.getPath());
    }

    @Test
    public void getLessonPathAfterChangeSectionId() throws Exception {
        assertEquals("section1/lesson1", lessonNode.getPath());
        sectionNode.setId(2);
        assertEquals("section2/lesson1", lessonNode.getPath());
    }

    @Test
    public void getStepPathAfterChangeLessonId() throws Exception {
        assertEquals("section1/lesson1/step1", stepNode.getPath());
        lessonNode.setId(2);
        assertEquals("section1/lesson2/step1", stepNode.getPath());
    }
}

File: stepik-union/src/test/java/org/stepik/core/courseFormat/StudyNodeTest.kt
package org.stepik.core.courseFormat

import org.junit.Assert.assertEquals
import org.junit.Before
import org.junit.Test


class StudyNodeTest {

    private var courseNode: CourseNode? = null
    private var sectionNode: SectionNode? = null
    private var lessonNode: LessonNode? = null
    private var stepNode: StepNode? = null

    @Before
    @Throws(Exception::class)
    fun setUp() {
        courseNode = CourseNode()

        sectionNode = SectionNode()
        sectionNode!!.id = 1
        sectionNode!!.parent = courseNode


        lessonNode = LessonNode()
        lessonNode!!.id = 1
        lessonNode!!.parent = sectionNode

        stepNode = StepNode()
        stepNode!!.id = 1
        stepNode!!.parent = lessonNode

        courseNode!!.children.add(sectionNode)
        sectionNode!!.children.add(lessonNode)
        lessonNode!!.children.add(stepNode)
    }

    @Test
    @Throws(Exception::class)
    fun getCourseDirectory() {
        assertEquals("", courseNode!!.directory)
    }

    @Test
    @Throws(Exception::class)
    fun getCoursePath() {
        assertEquals("", courseNode!!.path)
    }

    @Test
    @Throws(Exception::class)
    fun getSectionDirectory() {
        assertEquals("section1", sectionNode!!.directory)
    }

    @Test
    @Throws(Exception::class)
    fun getSectionPath() {
        assertEquals("section1", sectionNode!!.path)
    }

    @Test
    @Throws(Exception::class)
    fun getLessonDirectory() {
        assertEquals("lesson1", lessonNode!!.directory)
    }

    @Test
    @Throws(Exception::class)
    fun getLessonPath() {
        assertEquals("section1/lesson1", lessonNode!!.path)
    }

    @Test
    @Throws(Exception::class)
    fun getStepDirectory() {
        assertEquals("step1", stepNode!!.directory)
    }

    @Test
    @Throws(Exception::class)
    fun getStepPath() {
        assertEquals("section1/lesson1/step1", stepNode!!.path)
    }

    @Test
    @Throws(Exception::class)
    fun getSectionDirectoryAfterChangeId() {
        sectionNode!!.id = 2
        assertEquals("section2", sectionNode!!.directory)
    }

    @Test
    @Throws(Exception::class)
    fun getSectionPathAfterChangeId() {
        sectionNode!!.id = 2
        assertEquals("section2", sectionNode!!.path)
    }

    @Test
    @Throws(Exception::class)
    fun getLessonDirectoryAfterChangeId() {
        lessonNode!!.id = 2
        assertEquals("lesson2", lessonNode!!.directory)
    }

    @Test
    @Throws(Exception::class)
    fun getLessonPathAfterChangeId() {
        lessonNode!!.id = 2
        assertEquals("section1/lesson2", lessonNode!!.path)
    }

    @Test
    @Throws(Exception::class)
    fun getStepDirectoryAfterChangeId() {
        stepNode!!.id = 2
        assertEquals("step2", stepNode!!.directory)
    }

    @Test
    @Throws(Exception::class)
    fun getStepPathAfterChangeId() {
        stepNode!!.id = 2
        assertEquals("section1/lesson1/step2", stepNode!!.path)
    }

    @Test
    @Throws(Exception::class)
    fun getLessonPathAfterChangeSectionId() {
        assertEquals("section1/lesson1", lessonNode!!.path)
        sectionNode!!.id = 2
        assertEquals("section2/lesson1", lessonNode!!.path)
    }

    @Test
    @Throws(Exception::class)
    fun getStepPathAfterChangeLessonId() {
        assertEquals("section1/lesson1/step1", stepNode!!.path)
        lessonNode!!.id = 2
        assertEquals("section1/lesson2/step1", stepNode!!.path)
    }
}
